libftl/charon_protocol.c

```diff
@@ -52,3 +60,61 @@ ftl_charon_response_code_t ftl_charon_read_response_code(const char * response_s
    /* Got an invalid or unknown response code */
    return FTL_CHARON_UNKNOWN;
  }
+
+unsigned char decode_hex_char(char c) {
+    if (c >= '0' && c <= '9') {
+        return c - '0';
+    }
+    if (c >= 'A' && c <= 'Z') {
+        return (c - 'A') + 10;
+    }
+    if (c >= 'a' && c <= 'z') {
+        return (c - 'a') + 10;
+    }
+
+    return 0;
+}
+
+int recv_all(int sock, char * buf, int buflen) {
+    int pos = 0;
+    while (pos == 0 || buf[pos - 1] != '\n') {
+        pos += recv(sock, buf + pos, buflen - pos, 0);
```

Comment:
Missing error checking here if recv fails and returns -1 (bad network connection or something)